### PR TITLE
don't start the threadloop until we have work to do

### PR DESCRIPTION
**Pull request by blampe**
@junchaowu @breerly @abhinav

### CODE REVIEW

File: tchannel/sync/client.py [hunk 1]
```diff
@@ -115,6 +114,9 @@ def _wrap(self, f):
         assert callable(f)
 
         def wrapper(*a, **kw):
+            if not self._threadloop.is_ready():
```
Comment (breerly) [inline tchannel/sync/client.py]: I'm not quite sure that this makes sense, Threadloop should block until ready.

Comment (breerly) [inline tchannel/sync/client.py]: Where were we calling `start` before?

Comment (breerly) [inline tchannel/sync/client.py]: Dope. I see what you did there. Yeah, this will probably work.

**Comment by breerly** [comment]
Lazy thread, I could dig that. :shipit:

**Comment by abhinav** [comment]
:+1:

File: tchannel/sync/client.py [hunk 2]
```diff
@@ -103,7 +103,6 @@ def __init__(
 
         )
         self._threadloop = threadloop or ThreadLoop()
```
Comment (yurishkuro) [inline tchannel/sync/client.py]: if the caller passes an externally created threadloop, maybe tchannel should not be trying to start it at all.

Comment (blampe) [inline tchannel/sync/client.py]: Yeah, this also fixes a bug where the caller could have passed a started thread.

**Pull request merged**